### PR TITLE
Refactor project pay form

### PR DESCRIPTION
**Pull request by tomquirk**
The Pay input was getting out of hand and impossible to maintain.

This PR:
- adds a new context, PayProjectFormContext.
- adds v1 and v2v3-specific PayProjectFormContext providers.
- Uses these version-specific contexts in the respective project pages.
- adds a new hook, usePayProjectForm.
- Uses the new hook where appropriate

There are no user-facing changes.

Other minor changes:
- remove isSingleColumnLayout prop for project page. We can compute this internally using `isPreviewMode`.
- rename `weightedRate` => `weightAmountPerbicent`
- rename `weightedAmount` => `weightAmountPermyriad`
- Minor NFT rewards cleanup

### CODE REVIEW

**Comment by tomquirk** [comment]
@johnnyd-eth can you pls review this one instead? It affects NFT rewards, so want to make sure nothings broken. Cheers.

File: src/components/v2v3/V2V3Project/V2V3Project.tsx [hunk 1]
```diff
@@ -237,30 +196,19 @@ export function V2V3Project({
...
-            PayButton={V2PayButton}
-            reservedRate={fundingCycleMetadata?.reservedRate.toNumber()}
-            weight={fundingCycle?.weight}
-            weightingFn={weightedAmount}
-            tokenSymbol={tokenSymbol}
-            tokenAddress={tokenAddress}
-            disabled={isPreviewMode || payIsDisabledPreV2Redeploy()}
-            isEligibleForNft={isEligibleForNft}
-          />
+          <V2V3PayProjectFormProvider>
```
Comment (johnnyd-eth) [inline src/components/v2v3/V2V3Project/V2V3Project.tsx:199]: **excellent**

**Comment by johnnyd-eth** [comment]
Awesome cleanup @tomquirk. Code is great. 

Noticed a regression in the NFT rewards no longer being linked to the pay input:
1. Clicking NFTs doesn't fill the pay input
2. Filling pay input doesn't select the appropriate NFT

https://user-images.githubusercontent.com/96150256/193432802-c79d68a0-992a-4573-8e37-1444a0879f60.mp4

**Comment by johnnyd-eth** [comment]
Other than that, all looking good

**Review by johnnyd-eth — CHANGES_REQUESTED** [review]
.

**Comment by tomquirk** [comment]
ah shiet, thanks @johnnyd-eth . Realized the problem. Fixed it 🏓

**Comment by tomquirk** [comment]
@johnnyd-eth tested v1, v2 and nft rewards pay, all looking gucci. gonna merge

**Pull request merged**